pkg/builder/builder.go

```diff
@@ -841,6 +862,133 @@ func makeScrutinizeInitContainers(vscr *v1beta1.VerticaScrutinize, vdb *vapi.Ver
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        nm.Name,
+			Namespace:   nm.Namespace,
+			Labels:      MakeLabelsForStsObject(vdb, sc),
+			Annotations: MakeAnnotationsForStsObject(vdb, sc),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: MakeDepSelectorLabels(vdb, sc),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      MakeLabelsForPodObject(vdb, sc),
+					Annotations: MakeAnnotationsForObject(vdb),
+				},
+				Spec: buildVProxyPodSpec(vdb, sc),
+			},
+			Replicas: &vproxySize,
+		},
+	}
+}
+
+// buildPodSpec creates a PodSpec for the deployment
+func buildVProxyPodSpec(vdb *vapi.VerticaDB, sc *vapi.Subcluster) corev1.PodSpec {
+	termGracePeriod := int64(vmeta.GetTerminationGracePeriodSeconds(vdb.Annotations))
+	return corev1.PodSpec{
+		NodeSelector:                  sc.NodeSelector,
+		Affinity:                      GetK8sAffinity(sc.Affinity),
+		Tolerations:                   sc.Tolerations,
+		ImagePullSecrets:              GetK8sLocalObjectReferenceArray(vdb.Spec.ImagePullSecrets),
+		Containers:                    []corev1.Container{makeVProxyContainer(vdb, sc)},
+		TerminationGracePeriodSeconds: &termGracePeriod,
+		ServiceAccountName:            vdb.Spec.ServiceAccountName,
+		SecurityContext:               vdb.Spec.PodSecurityContext,
+		Volumes: []corev1.Volume{
+			{
+				Name: sc.Name,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: sc.GetVProxyConfigMapName(vdb)},
+						Items: []corev1.KeyToPath{
+							{
+								Key:  "config.yaml",
+								Path: "config.yaml",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// makeDataForProxyConfigMap generates a configmap data in config.yaml format
+func makeDataForVProxyConfigMap(vdb *vapi.VerticaDB, sc *vapi.Subcluster) string {
+	var nodeList []string
+	port := 5433
+
+	for scIndex := int32(0); scIndex < sc.Size; scIndex++ {
+		nodeItem := fmt.Sprintf("%s-%s-%d:%d", vdb.Name, sc.Name, scIndex, port)
+		nodeList = append(nodeList, nodeItem)
+	}
+
+	proxyData := ProxyData{
+		Listener: map[string]interface{}{
+			"host": "",
+			"port": port,
+		},
+		Database: map[string][]string{
+			"nodes": nodeList,
+		},
+		Log: map[string]string{
+			"level": "INFO",
+		},
+	}
+
+	pData, _ := yaml.Marshal(proxyData)
+	return string(pData)
+}
+
+// BuildVProxyConfigMap builds a config map for client proxy
+func BuildVProxyConfigMap(nm types.NamespacedName, vdb *vapi.VerticaDB, sc *vapi.Subcluster) *corev1.ConfigMap {
+	immutable := false
+	proxyConfig := makeDataForVProxyConfigMap(vdb, sc)
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            nm.Name,
+			Namespace:       nm.Namespace,
+			Labels:          MakeLabelsForPodObject(vdb, sc),
```

Comment:
```suggestion
			Labels:          MakeLabelsForVProxyConfigMap(vdb, sc),
```

Reply:
Thanks. Added MakeLabelsForVProxyObject to add labels for both proxy pods and configmaps.